app/src/main/java/ch/epfl/sdp/drone3d/service/storage/dao/FirebaseMappingMissionDao.kt

```diff
@@ -128,10 +140,76 @@ class FirebaseMappingMissionDao @Inject constructor(
         return getMappingMissions(null)
     }
 
+    override fun getPrivateFilteredMappingMissions(): LiveData<List<MappingMission>> {
+        return privateFilteredMappingMissions
+    }
+
+    override fun getSharedFilteredMappingMissions(): LiveData<List<MappingMission>> {
+        return sharedFilteredMappingMissions
+    }
+
+    /**
+     * Update the list of mapping missions filtered by the new [filter]
+     * If the [ownerUid] is not null the missions are fetch from private repo of the owner
+     * Otherwise in the shared repo
+     */
+    private fun updateFilteredMappingMissions(ownerUid: String?, filter: String) {
+        val rootRef =
+            if (ownerUid != null) privateMappingMissionRef(ownerUid) else sharedMappingMissionRef()
+        val mappingMissions =
+            if (ownerUid != null) privateFilteredMappingMissions else sharedFilteredMappingMissions
+        val query =
+            if (ownerUid != null) privateFilteredMappingMissionsQuery else sharedFilteredMappingMissionsQuery
+        val listener =
+            if (ownerUid != null) privateFilteredMappingMissionsListener else sharedFilteredMappingMissionsListener
+
+        if (query != null && listener != null) {
+            query.removeEventListener(listener)
+        }
+
+        val missionsQuery = rootRef.orderByChild("nameUpperCase").startAt(filter).endAt(filter + "\uf8ff")
```

Comment:
What is "\uf8ff" ?

Reply:
\uf8ff is the last unicode character. I use it so the query will get all missions that start with the substring filter